### PR TITLE
net/gcoap: do not allocate RX buf on stack

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
Currently, `gcoap`s internal receive buffer is allocated on the stack. For the default size of 128 bytes this might be ok (although borderline I'd say), but if one decides to increase that value it soon becomes dangerous.

This PR simply moves the receive buffer from the stack into the `bss` section of memory, decoupling the stacksize from the buffer size.

Background: I am currently doing some floodtesting using different high-level protocols on top of BLE, using payload sizes of [2:1200] bytes. So when setting `GCOAP_PDU_BUF_SIZE` to 1250, the current state of the code does not like that very much (-> stack overflow).

### Testing procedure
Simply verify that `examples/gcoap` is working as before, in the default configuration this change should not have any runtime impacts, the only thing that changes is a lower stack usage for the `gcoap` thread in exchange for an increased `bss` section size.

### Issues/PRs references
none

### CODE REVIEW

File: sys/net/application_layer/gcoap/gcoap.c [hunk 1]
```diff
@@ -213,9 +213,9 @@ static void _listen(sock_udp_t *sock)
     case COAP_CLASS_REQ:
         if (coap_get_type(&pdu) == COAP_TYPE_NON
                 || coap_get_type(&pdu) == COAP_TYPE_CON) {
-            size_t pdu_len = _handle_req(&pdu, buf, sizeof(buf), &remote);
+            size_t pdu_len = _handle_req(&pdu, _rxbuf, sizeof(_rxbuf), &remote);
             if (pdu_len > 0) {
-                ssize_t bytes = sock_udp_send(sock, buf, pdu_len, &remote);
+                ssize_t bytes = sock_udp_send(sock, _rxbuf, pdu_len, &remote);
```
Comment (miri64) [inline sys/net/application_layer/gcoap/gcoap.c]: Nitpick: it's a bit confusing, that `_rxbuf` is used here for sending.

**Comment by haukepetersen** [comment]
changed the name of the buffer variable to `_listen_buf` (squashed right in)

**Comment by haukepetersen** [comment]
next time I call it `_mathilda` :-)

**Comment by miri64** [comment]
`_navi` https://www.youtube.com/watch?v=mD4pVIMrDmg

**Comment by haukepetersen** [comment]
All green -> go.

**Pull request merged**